### DISCUSSION TITLE
Virtual RMU

### DISCUSSION THREAD

**Discussion by humbeecc** [yozik04/nibe#58] [General]
First of all congrats with this great component!

Not sure if this is the good place, but is there some functionality foreseen to use (or misuse) a virtual RMU? Or is it already possible?
I know that NibePI has something like that. It allows for example to use a non-Nibe indoor temperature sensor (KNX or whatever) to be used as a room sensor.

**Comment by elupus** [comment]
Its in planning. Not decided how yet. The problem is that the original nibegw arduino/pi code does not support it. So it needs an extension. I can add that to esphome-nibe, but id like to stay compatible if possible.

So may i should just add that support fully in esphome-nibe. But.. that would complicate that device. So not really sure what to do yet. But its planned!

**Reply by teixemf** [reply]
Would it be possible to support for the S-series via TCP Modbus?

**Reply by elupus** [reply]
Not as far as i know. You could ask NIBE if they support RMU:s connected via modbus for F series. Maybe they have planned support for that.

**Comment by humbeecc** [comment]
Great that it is planned.
I understand the problem. At this time nibegw is indeed only a "gateway", without additional functionality (based on addresses). That's the beauty as well as the limitation of it. That's probably also the reason that for example resetting the modbus alarm by itself is not handled directly by nibegw, but needs to be done by its clients (I use some Home Assistant automation for that)
